docs/release-notes/searcher-context-removal.rst

```diff
@@ -0,0 +1,67 @@
+:orphan:
+
+**Breaking Changes**
+
+-  ASHA: All experiments using ASHA hyperparameter search must now configure ``max_time`` and
+   ``time_metric`` in the experiment config, instead of ``max_length``. Additionally, training code
+   must report the configured ``time_metric`` in validation metrics. As a convenience, Determined
+   training loops now automatically report ``batches`` and ``epochs`` with metrics, which you can
+   use as your ``time_metric``. ASHA experiments without this modification will no longer run.
+
+-  Custom Searchers: all custom searchers including DeepSpeed Autotune were deprecated in ``0.36.0``
+   and are now being removed. Users are encouraged to use a preset searcher, which can be easily
+   :ref:`configured <experiment-configuration_searcher>` for any experiment.
+
+**New Features**
+
+-  API: introduce ``keras.DeterminedCallback``, a new high-level training API for TF Keras that
+   integrates Keras training code with Determined through a single :ref:`Keras Callback
+   <api-keras-ug>`.
+
+-  API: introduce ``deepspeed.Trainer``, a new high-level training API for DeepSpeedTrial that
```

Comment:
One of the changes to `DeepSpeedTrialContext` was to remove `det.EnvContext` from it. This used to be accessible via `context.env`. I mentioned this as a potential breaking change for some users, who may have to move to the `context.get_experiment_config()` and `context.get_hparams()` methods.

Reply:
Actually, should probably just tell users it's gone completely and they should read config from the cluster info.

Reply:
This shouldn't technically be breaking because it was never documented, so it was not technically public.

But it's so old that it wasn't prefixed with `_` because we didn't used to be good about doing that.

Reply:
If it wasn't documented before, we don't need to document that it is gone, imo.

Reply:
The reason I reference it is because our examples had code that used it (`gpt_neox` before the Trainer API rewrites, for example)